packages/aws-cdk-lib/aws-sns/lib/subscription.ts

```diff
@@ -152,10 +160,85 @@ export class Subscription extends Resource {
+      const minDelayTarget = healthyRetryPolicy.minDelayTarget;
+      const maxDelayTarget = healthyRetryPolicy.maxDelayTarget;
+      if (minDelayTarget !== undefined) {
+        if (minDelayTarget.toMilliseconds() % 1000 !== 0) {
+          throw new Error(`minDelayTarget must be a whole number of seconds, got: ${minDelayTarget}`);
+        }
+        const minDelayTargetSecs = minDelayTarget.toSeconds();
+        if (minDelayTargetSecs < 1 || minDelayTargetSecs > delayTargetLimitSecs) {
+          throw new Error(`minDelayTarget must be between 1 and ${delayTargetLimitSecs} seconds inclusive, got: ${minDelayTargetSecs}s`);
+        }
+      }
+      if (maxDelayTarget !== undefined) {
+        if (maxDelayTarget.toMilliseconds() % 1000 !== 0) {
+          throw new Error(`maxDelayTarget must be a whole number of seconds, got: ${maxDelayTarget}`);
+        }
+        const maxDelayTargetSecs = maxDelayTarget.toSeconds();
+        if (maxDelayTargetSecs < 1 || maxDelayTargetSecs > delayTargetLimitSecs) {
+          throw new Error(`maxDelayTarget must be between 1 and ${delayTargetLimitSecs} seconds inclusive, got: ${maxDelayTargetSecs}s`);
+        }
+        if ((minDelayTarget !== undefined) && minDelayTarget.toSeconds() > maxDelayTargetSecs) {
+          throw new Error('minDelayTarget must not exceed maxDelayTarget');
+        }
+      }
+
+      const numRetriesLimit = 100;
+      if (healthyRetryPolicy.numRetries && (healthyRetryPolicy.numRetries < 0 || healthyRetryPolicy.numRetries > numRetriesLimit)) {
+        throw new Error(`numRetries must be between 0 and ${numRetriesLimit} inclusive, got: ${healthyRetryPolicy.numRetries}`);
+      }
+      const { numNoDelayRetries, numMinDelayRetries, numMaxDelayRetries } = healthyRetryPolicy;
+      if (numNoDelayRetries && (numNoDelayRetries < 0 || !Number.isInteger(numNoDelayRetries))) {
+        throw new Error(`numNoDelayRetries must be an integer zero or greater, got: ${numNoDelayRetries}`);
+      }
+      if (numMinDelayRetries && (numMinDelayRetries < 0 || !Number.isInteger(numMinDelayRetries))) {
+        throw new Error(`numMinDelayRetries must be an integer zero or greater, got: ${numMinDelayRetries}`);
+      }
+      if (numMaxDelayRetries && (numMaxDelayRetries < 0 || !Number.isInteger(numMaxDelayRetries))) {
+        throw new Error(`numMaxDelayRetries must be an integer zero or greater, got: ${numMaxDelayRetries}`);
+      }
+    }
+    if (throttlePolicy) {
+      const maxReceivesPerSecond = throttlePolicy.maxReceivesPerSecond;
+      if (maxReceivesPerSecond !== undefined && (maxReceivesPerSecond < 1 || !Number.isInteger(maxReceivesPerSecond))) {
+        throw new Error(`maxReceivesPerSecond must be an integer greater than zero, got: ${maxReceivesPerSecond}`);
+      }
+    }
+    return {
+      healthyRetryPolicy: healthyRetryPolicy ? {
+        // minDelayTarget, maxDelayTarget and numRetries are (empirically) mandatory when healthyRetryPolicy is specified,
```

Comment:
can you explain what you meant by `empirically mandatory)`? I do see that they do have default values set.

Reply:
Does https://github.com/aws/aws-cdk/pull/30830#discussion_r1770405277 answer your question (also see last comment in that thread as well as to why they're optional)?